yaptide/celery/tasks.py

```diff
@@ -0,0 +1,181 @@
+from yaptide.celery.worker import celery_app
+
+import os
+import sys
+import tempfile
+
+from celery import states
+from celery import exceptions as celery_exceptions
+from celery.result import AsyncResult
+
+from pymchelper.executor.options import SimulationSettings
+from pymchelper.executor.runner import Runner as SHRunner
+from pymchelper.estimator import Estimator
+from pymchelper.page import Page
+from pymchelper.axis import MeshAxis
+
+# dirty hack needed to properly handle relative imports in the converter submodule
+sys.path.append('yaptide/converter')
+from ..converter.converter.api import get_parser_from_str, run_parser  # skipcq: FLK-E402
+
+
+@celery_app.task(bind=True)
+def run_simulation(self, param_dict: dict, raw_input_dict: dict):
+    """Simulation runner"""
+    # create temporary directory
+    with tempfile.TemporaryDirectory() as tmp_dir_path:
+
+        # digest dictionary with project data (extracted from JSON file)
+        # and generate SHIELD-HIT12A input files
+        conv_parser = get_parser_from_str(param_dict['sim_type'])
+        run_parser(parser=conv_parser, input_data=raw_input_dict, output_dir=tmp_dir_path)
+
+        settings = SimulationSettings(input_path=tmp_dir_path,  # skipcq: PYL-W0612
+                                      simulator_exec_path=None,
+                                      cmdline_opts='')
+
+        runner_obj = SHRunner(jobs=param_dict['jobs'],
+                              keep_workspace_after_run=True,
+                              output_directory=tmp_dir_path)
+
+        self.update_state(state="PROGRESS", meta={"path": tmp_dir_path})
+        is_run_ok = runner_obj.run(settings=settings)
+        if not is_run_ok:
+            self.update_state(state=states.FAILURE)
+            raise celery_exceptions.TaskError
+
+        estimators_dict: dict = runner_obj.get_data()
+
+        result: dict = dummy_convert_output(estimators_dict)
+
+        return {'status': 'COMPLETED', 'result': result}
+
+
+def dummy_convert_output(estimators_dict: dict) -> dict:
+    """Dummy function for converting simulation output to dictionary"""
+    if not estimators_dict:
+        return {'message': 'No estimators'}
+
+    # result_dict is the dictionary object, which is later converted to json
+    # to provide readable api response for fronted
+
+    # result_dict contains the list of estimators
+    result_dict = {'estimators': []}
+    estimator: Estimator
+    for estimator_key, estimator in estimators_dict.items():
+        # est_dict contains list of pages
+        est_dict = {
+            'name': estimator_key,
+            'pages': [],
+            }
+
+        page: Page   # type is still marked as Never
+        for page in estimator.pages:
+            # currently we are handling for sure only 1-D results
+            # 0-D and 2-D results aren't tested yet, due to testing problems
```

Comment:
what kind of testing problems do we have ?

Reply:
Testing requires modyfing converter files, which we didn't want to do, that's why they haven't been tested yet, but tomorrow we plan to do it anyway due to little amount time left